### DISCUSSION TITLE
Multiple regions and information about update status

### DISCUSSION THREAD

**Discussion by arthurwozniak** [osm-search/Nominatim#2127] [Q&A]
Hi,
we have installed nominatim with multiple regions option (https://nominatim.org/release-docs/3.5.1/admin/Advanced-Installations/#updating-multiple-regions). Updates seems to be working and `sequence.state` is growing. But confusing for me is that no information is available via API/nominatim-ui.

When I take a look inside UI js fetching last update date (https://github.com/osm-search/nominatim-ui/blob/8c17a599828a55c8017d60c49d02305424a9ee7e/src/assets/js/base.js#L129), it uses function `dataDate` from https://github.com/osm-search/Nominatim/blob/master/lib/Status.php.

Problem is that `import_status` table is updated only in file https://github.com/osm-search/Nominatim/blob/f3e0e401fd50dd48eaba93ffdc322f3c3516460e/utils/update.php, specifically using arguments `--init-updates` or `--import-osmosis*`.

But updating multiple regions calls uses `--import-diff` arg (https://github.com/osm-search/Nominatim/blob/master/utils/update_database.sh).

So one can end in situation that data are updating for year, but `Data last updated:` is invalid and confusing. Is it correct or do I oversee something?

**Comment by lonvia** [comment]
Your observation is correct. The `status` table is usually updated by `update.php --import-osmosis` or `update.php --import-osmosis-all`. Neither function is called from the `update_database.sh` script. It is save to set the data in the table manually at the end of `update_database.sh`. Should be fairly easy to do using `psql` and the date from one of the sequence files. Feel free to send in a PR if you have a good solution.

**Reply by arthurwozniak** [reply]
Thanks for confirmation of our thoughts! Well, honestly I consider fix more complex as it may seem from beginning. In my opinion it is necessary to store last update for each region and then show last update date per region in UI. It would require modifications in API, FE, etc. Because last update of e.g. Germany does not mean that France is up to date too.

**Reply by lonvia** [reply]
That would be a bit special. Nominatim doesn't internally have a notion of datasources and there are no plans to add that. Each data object remembers when it was updated the last time. If you really need a fine granularity that is likely the better place to start. It better reflects what Nominatim has actually been updating. There are a couple of short cuts in the code, where dependent data updates are skipped for performance reasons.
